CMakeLists.txt

```diff
@@ -414,18 +414,35 @@ if (COOLPROP_VXWORKS_LIBRARY_MODULE OR COOLPROP_VXWORKS_LIBRARY)
     install (TARGETS ${app_name} DESTINATION "${COOLPROP_INSTALL_PREFIX}/shared_library/VxWorks")
 endif()
 
-if (COOLPROP_MATHCAD_MODULE)
+if (COOLPROP_PRIME_MODULE)
     if(NOT WIN32)
-      message(FATAL_ERROR "COOLPROP_MATHCAD_MODULE can only be used on windows host")
+      message(FATAL_ERROR "COOLPROP_PRIME_MODULE can only be used on windows host")
     endif()
-    IF( "${COOLPROP_MATHCAD_ROOT}" STREQUAL "")
-        message(FATAL_ERROR "You must provide the path to MathCAD Root directory using something like -DCOOLPROP_MATHCAD_ROOT=\"C:/Program Files/PTC/Mathcad Prime 3.0\"")
+    IF( "${COOLPROP_PRIME_ROOT}" STREQUAL "")
+        message(FATAL_ERROR "You must provide the path to MathCAD Prime Root directory using something like -DCOOLPROP_PRIME_ROOT=\"C:/Program Files/PTC/Mathcad Prime 3.1\"")
     endif()
     list(APPEND APP_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/src/CoolPropLib.cpp")
     list(APPEND APP_SOURCES "wrappers/MathCAD/CoolPropMathcad.cpp")
     add_library(CoolPropMathcadWrapper SHARED ${APP_SOURCES})
-    include_directories("${COOLPROP_MATHCAD_ROOT}/Custom Functions")
-    target_link_libraries(CoolPropMathcadWrapper "${COOLPROP_MATHCAD_ROOT}/Custom Functions/mcaduser.lib")
+    include_directories("${COOLPROP_PRIME_ROOT}/Custom Functions")
+    target_link_libraries(CoolPropMathcadWrapper "${COOLPROP_PRIME_ROOT}/Custom Functions/mcaduser.lib")
+    SET_TARGET_PROPERTIES(CoolPropMathcadWrapper PROPERTIES LINK_FLAGS "/ENTRY:\"DllEntryPoint\"")
+    add_dependencies (CoolPropMathcadWrapper generate_headers)
+    set_target_properties(CoolPropMathcadWrapper PROPERTIES SUFFIX ".dll" PREFIX "")
+endif()
+
+if (COOLPROP_MATHCAD15_MODULE)
+    if(NOT WIN32)
+      message(FATAL_ERROR "COOLPROP_MATHCAD15_MODULE can only be used on windows host")
+    endif()
+    IF( "${COOLPROP_MATHCAD15_ROOT}" STREQUAL "")
+        message(FATAL_ERROR "You must provide the path to MathCAD 15 Root directory using something like -DCOOLPROP_MATHCAD15_ROOT=\"C:/Program Files (x86)/Mathcad/Mathcad 15\"")
+    endif()
+    list(APPEND APP_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/src/CoolPropLib.cpp")
+    list(APPEND APP_SOURCES "wrappers/MathCAD/CoolPropMathcad.cpp")
```

Comment:
Paths that are relative to the root where CMakeLists.txt sits should have ``${CMAKE_CURRENT_SOURCE_DIR}`` prepended like the line above